### PR TITLE
feat: filters preset

### PR DESCRIPTION
**Pull request by topliceanurazvan**
## Changes

- Save filter preset 
- Load filter preset 
- Delete filter preset
- Refactor resource filters

## Screenshots

image

image

image

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/molecules/ResourceFilter/ResourceFilter.tsx [hunk 1]
```diff
@@ -187,12 +179,32 @@ const ResourceFilter = () => {
 
   return (
     <S.Container>
-      <S.Title>
-        <S.TitleLabel>Filter resources by:</S.TitleLabel>
-        <S.TitleButton type="link" onClick={resetFilters} disabled={areFiltersDisabled}>
-          Reset all
-        </S.TitleButton>
-      </S.Title>
+      <S.TitleContainer>
```
Comment (WitoDelnat) [inline src/components/molecules/ResourceFilter/ResourceFilter.tsx:182]: Nitpick but the UI does not look good if you put navigator on minimum width.

I also find 'Load preset' and 'Save preset' a bit verbose but that's a bit personal.

Screenshot 2022-05-25 at 12 33 40

**Comment by WitoDelnat** [comment]
There is a problem with styling the background as seen in the screenshot:

Screenshot 2022-05-25 at 12 37 11

**Comment by WitoDelnat** [comment]
The kind selector is broken as seen in this recording

https://user-images.githubusercontent.com/7761005/170243808-a801cfdb-b492-4bba-8156-c3176b16ce35.mov

**Comment by WitoDelnat** [comment]
A bit a product nitpick but does it make sense to allow them to "Save" an empty filter?

File: src/components/molecules/ResourceFilter/ResourceFilter.tsx (hunk 1 above)
Comment (devcatalin) [reply src/components/molecules/ResourceFilter/ResourceFilter.tsx:192]: We could show "Load" and "Save" when there's not enough space

File: src/components/molecules/ResourceFilter/ResourceFilter.tsx [hunk 2]
```diff
@@ -1,14 +1,19 @@
 import {useEffect, useMemo, useState} from 'react';
 import {useDebounce} from 'react-use';
 
-import {Input, Select} from 'antd';
+import {Button, Input, Select, Tooltip} from 'antd';
+
+import {ClearOutlined} from '@ant-design/icons';
 
 import {mapValues} from 'lodash';
+import isEmpty from 'lodash/isEmpty';
```
Comment (devcatalin) [inline src/components/molecules/ResourceFilter/ResourceFilter.tsx:9]: why not add this to the import from above?

File: src/components/organisms/FiltersPresetModal/FiltersPresetModal.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import {AlertEnum} from '@models/alert';
+
+import {useAppDispatch, useAppSelector} from '@redux/hooks';
+import {setAlert} from '@redux/reducers/alert';
+import {deleteFilterPreset, loadFilterPreset, saveFilterPreset} from '@redux/reducers/main';
+import {closeFiltersPresetModal} from '@redux/reducers/ui';
+
+import * as S from './FiltersPresetModal.styled';
+
+const FiltersPresetModal: React.FC = () => {
```
Comment (devcatalin) [inline src/components/organisms/FiltersPresetModal/FiltersPresetModal.tsx:14]: I think I would rather show directly a dropdown with search when the `Load preset` button is clicked.
Alternatively, make the dropdown in this modal to automatically expand and set it to auto focus.

File: src/components/organisms/NavigatorPane/NavigatorPane.tsx [hunk 4]
```diff
@@ -115,7 +115,7 @@ const NavPane: React.FC<Props> = ({height}) => {
 
       <ReflexContainer orientation="horizontal" style={{height: height - 40}}>
         {isResourceFiltersOpen && (
-          <ReflexElement flex={0.4} minSize={100}>
+          <ReflexElement style={{background: '#0c0d0e'}} flex={0.4} minSize={100}>
```
Comment (devcatalin) [inline src/components/organisms/NavigatorPane/NavigatorPane.tsx:118]: maybe this should be added to the `@styles/Colors`

File: src/constants/tooltips.ts [hunk 5]
```diff
@@ -69,3 +69,4 @@ export const KustomizeTabTooltip = `View Kustomizations (${KEY_CTRL_CMD}+SHIFT+K
 export const HelmTabTooltip = `View Helm Charts (${KEY_CTRL_CMD}+SHIFT+H)`;
 export const TemplatesTabTooltip = `View Templates`;
 export const ValidationTabTooltip = `View Validation (${KEY_CTRL_CMD}+SHIFT+V)`;
+export const ResetFiltersTooltip = `Reset Filters (${KEY_CTRL_CMD}+ALT+R)`;
```
Comment (devcatalin) [inline src/constants/tooltips.ts]: is this a standard tooltip to reset filters or a search query?

File: src/index.css [hunk 6]
```diff
@@ -369,3 +369,13 @@ div.monokleEditorErrorRefGlyphClass.monokleEditorUnsatisfiedRefGlyphClass::after
 ::-webkit-scrollbar-thumb:hover {
   background: #242424;
 }
+
+.filters-preset-dropdown .ant-select-item-option-content {
```
Comment (devcatalin) [inline src/index.css:373]: couldn't this be done in a styled component, or is it because the dropdown is rendered outside the component tree?

**Review by devcatalin — CHANGES_REQUESTED** [review]

File: src/index.css (hunk 6 above)
Comment (topliceanurazvan) [reply src/index.css:373]: Exactly, couldn't be done with a styled component

File: src/constants/tooltips.ts (hunk 5 above)
Comment (topliceanurazvan) [reply src/constants/tooltips.ts]: It's just a tooltip for the reset filters icon

**Review by devcatalin — APPROVED** [review]

**Pull request merged**